### ISSUE TITLE
2.5.22294: GameParser#parseInfo:319 - java.lang.NullPointerException

### ISSUE THREAD

**Issue by tripleabuilderbot** [triplea-game/triplea#8086]
## User Description
when I select a map a error popup comes up also the preview pictures of the maps wont come up instead a 
message saying error reading file 

## Log Message
Could not parse:jar:file:/C:/Users/hp/triplea/downloadedMaps/classic_variations-master.zip!/classic_variations-master/map/games/gold.xml, null

## TripleA Version
2.5.22294

## Java Version
11.0.6

## Operating System
Windows 10

## Memory
Heap utilization statistics [MB]
Used Memory: 29
Free memory: 1950
Total memory: 1979
Max memory: 1979

## Stack Trace
```
Exception: java.lang.NullPointerException
java.lang.Exception
	at games.strategy.engine.data.gameparser.GameParser.parseInfo(GameParser.java:319)
	at games.strategy.engine.data.gameparser.GameParser.parse(GameParser.java:129)
	at games.strategy.engine.data.gameparser.GameParser.lambda$parse$0(GameParser.java:110)
	at org.triplea.java.UrlStreams.openStream(UrlStreams.java:73)
	at games.strategy.engine.data.gameparser.GameParser.parse(GameParser.java:105)
	at games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel.lambda$new$0(GameSelectorModel.java:59)
	at games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel.parseAndValidate(GameSelectorModel.java:126)
	at games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel.load(GameSelectorModel.java:99)
	at games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorPanel.lambda$selectGameFile$8(GameSelectorPanel.java:399)
	at games.strategy.engine.framework.ui.background.BackgroundTaskRunner.lambda$runInBackground$0(BackgroundTaskRunner.java:45)
	at games.strategy.engine.framework.ui.background.BackgroundTaskRunner$1.doInBackground(BackgroundTaskRunner.java:108)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)


```

**Comment by DanVanAtta** [comment]
@Navalland2  - Try deleting: C:/Users/hp/triplea/downloadedMaps/classic_variations-master.zip

**Comment by DanVanAtta** [comment]
@Navalland2  per our DM message, if you are seeing this for any map, verify that the XML contains a `<info version=..' attribute. You'll need to crack open one of the zip files and inspect the XML reported to have a problem to check that.

**Comment by Navalland2** [comment]
I don't have classic_variations, I even deleted classic but the issue stil present.

**Comment by DanVanAtta** [comment]
Also, check the repeated error messages, is it the same map that is being reported as a problem. Do try and delete (or move) the map reported as a problem and let's go from there.

**Comment by DanVanAtta** [comment]
It is this file: C:/Users/hp/triplea/downloadedMaps/classic_variations-master.zip

Check the error message details to see if a different file is being reported. If that file is gone it'll be interesting to see the full stack trace from the error with that file deleted.

**Comment by Navalland2** [comment]
I think its not related with maps. I've even removed everything from downloadedMaps but Feudal Japan as example but it didn't help either.

**Comment by DanVanAtta** [comment]
Can you post the stack trace from that case @Navalland2 ? When you click 'submit error report', there is a 'preview' button that has the stack trace.

**Comment by Navalland2** [comment]
I don't know how could there be preview button. The error lists as gave is exactly;

https://i.ibb.co/B662cGt/errors.png

**Comment by DanVanAtta** [comment]
Ah, thanks for the screenshot. You'll need to fish out the stack trace from the 'console'. To open that, it is in the 'settings' menu, 'game' tab, 'show console -> on'. It should display the latest error message.

**Comment by Navalland2** [comment]
List of ÖÖ errors for each of every maps than final error message.

https://i.ibb.co/GHvKJVw/error2.png

**Comment by DanVanAtta** [comment]
The 'INFO' messages are fine. The last two 'SEVERE' error messages are not, looks like there are two map zips that have problems. First try moving those two out of your downloads folder, that will help isolate the problem.

**Comment by DanVanAtta** [comment]
FYI, if it is any easier, large text outputs can be shared by creating a 'gist' https://gist.github.com/

**Comment by Navalland2** [comment]
This version after ÖÖ errors...

https://i.ibb.co/WKb6Ntj/error4.png

**Comment by DanVanAtta** [comment]
Might be easiest to rename the 'downloadMaps' folder. Download a map, see if it works.

**Comment by Navalland2** [comment]
It didn't work either.

**Comment by DanVanAtta** [comment]
After removing all maps from 'downloadedMaps', and downloading a new map, you are still seeing a 'Severe' error message for the map you just downloaded?

**Comment by Navalland2** [comment]
Yes, for example if i remove all of them and instead download tutorial.

https://i.ibb.co/xh4xvYN/error6.png

**Comment by DanVanAtta** [comment]
Have you tried running TripleA as administrator? @Navalland2 
I'm thinking there must be an earlier error, reported or silent, that is preventing TripleA from reading the file at all.

**Comment by Navalland2** [comment]
I had tried to run TripleA as administrator. It didn't solve the issue.

BTW I can open load games in 2.5.

**Comment by DanVanAtta** [comment]
Load games do not do XML parsing, which is where the problem is - so that makes sense.

@Navalland2 it's hard to know what the problem is, it is not systemic as this does work on other systems. Can you see if upgrading the installed version of Java you have makes any difference, bit of a long shot.

**Comment by Navalland2** [comment]
Still same unfortunately. What were the major changes after 2.2 which might be real reason? I feel it could be something to do with language codes since "Ö" is one of the exclusive letter in my language. Or "ÖÖ" represent more different thing.

**Comment by DanVanAtta** [comment]
The character encoding in the log is not going to be related.


In 2.3 we switched XML parsers: https://triplea-game.org/release_notes/

> #7596 | UPDATE | Map XML parser re-written to allow for more flexible XML structure. The XML DTD is no longer checked or used.

What is the exact java version that the engine is reporting when you launch the game?

**Comment by Navalland2** [comment]
Currently I have only Java(TM) platform SE binary

Version: 8.0.2710.9

They are Java related error reports when -for example- I try to open Camp David.

SEVERE: Could not parse:jar:file:/C:/Users/Acer/triplea/downloadedMaps/camp_david-master.zip!/camp_david-master/map/games/1973_Yom_Kippur_War.xml, null
java.lang.NullPointerException
	at games.strategy.engine.data.gameparser.GameParser.parseInfo(GameParser.java:319)
	at games.strategy.engine.data.gameparser.GameParser.parse(GameParser.java:129)
	at games.strategy.engine.data.gameparser.GameParser.lambda$parse$0(GameParser.java:110)
	at org.triplea.java.UrlStreams.openStream(UrlStreams.java:73)
	at games.strategy.engine.data.gameparser.GameParser.parse(GameParser.java:105)
	at games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel.lambda$new$0(GameSelectorModel.java:59)
	at games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel.parseAndValidate(GameSelectorModel.java:126)
	at games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel.load(GameSelectorModel.java:99)
	at games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorPanel.lambda$selectGameFile$8(GameSelectorPanel.java:399)
	at games.strategy.engine.framework.ui.background.BackgroundTaskRunner.lambda$runInBackground$0(BackgroundTaskRunner.java:45)
	at games.strategy.engine.framework.ui.background.BackgroundTaskRunner$1.doInBackground(BackgroundTaskRunner.java:108)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

**Comment by DanVanAtta** [comment]
Java 8 is getting to be pretty old. It should mean that you are using the bundled java as TripleA needs Java11 or better.

Could you check if a latest pre-release works @Navalland2 ?

https://github.com/triplea-game/triplea/releases

The bundled java version has been updated to a more recent version not too long ago. Failing any other leads, it's worth trying at this point.

**Comment by Navalland2** [comment]
I have downloaded jdk-11.0.9. 

In TripleA 2.2 it is seen that I am using java 11.0.6 while in the latest version 2.6 it is seen 11.0.9.1.

I can't open maps in 2.6 either.

**Comment by DanVanAtta** [comment]
You'll see this problem in 2.3+ @Navalland2 
The JDK change was a longshot. I think we're going to have to add in a fair amount of verbose debug information to determine where things are going wrong and will need you to send that debug information back. The process will involve downloading a latest pre-release, installing, launching the game, hitting an error, then finding the log file, then replying back here and attaching that log file to your response. Would you be willing to help with that @Navalland2 ? 

At some point we will require players to be on version 2.5 of greater to join the lobby, it is important we get this resolved.

**Comment by Navalland2** [comment]
Of course, I would want to do everything to solve the issue.

**Comment by Navalland2** [comment]
..........

**Comment by DanVanAtta** [comment]
Stay tuned @Navalland2 , I'm trying to wrap up the `map.yml` project first and then will come to this issue.

**Comment by DanVanAtta** [comment]
Let's move the debug conversation to: https://github.com/triplea-game/triplea/issues/8711
I'd like to do so for the issue to be easier to find.

@Navalland2 please check that issue if you would and thank you in advance for the help debugging, hopefully we can get this solved together and you on the latest version of TripleA.

**Comment by DanVanAtta** [comment]
This should be fixed actually in 2.6

For downloaded maps, that is a different issue and let's track with its own bug report.
